src/fiat/fiat.ts

```diff
@@ -1,5 +1,39 @@
 const numSatsInBtc = 100_000_000;
 
+interface CurrencyDetails {
```

Comment:
I don't think we need all these properties. Can you check how this is done in [BuzzPay](http://github.com/getAlby/pos) and [Alby Go](https://github.com/getAlby/go) to see what's needed and ensure we can remove the duplicated code there and use this new method instead?